src/logservice/logminer/ob_log_miner_logger.cpp

```diff
@@ -50,25 +52,40 @@ int LogMinerLogger::log_progress(int64_t record_num, int64_t current_ts, int64_t
   int ret = OB_SUCCESS;
   double percentage = double(current_ts - begin_ts) / double(end_ts - begin_ts);
   double progress = 0;
+  struct winsize w;
+  int pb_width = 0;
   int lpad = 0;
   int rpad = 0;
   int64_t pos = 0;
   const ObString nls_format;
   char time_buf[128] = {0};
+  char pb_buf[MAX_SCREEN_WIDTH] = {0}; 
   // current_ts may exceed end_ts
   if (percentage > 1) {
     percentage = 1;
   }
   progress = percentage * 100;
-  lpad = (int)(percentage * PB_WIDTH);
-  rpad = PB_WIDTH - lpad;
+
+  if (!ioctl(STDOUT_FILENO, TIOCGWINSZ, &w)) { // get the width of terminal window
```

Comment:
What happens if the user redirects the standard output to a file?

Reply:
https://github.com/oceanbase/oceanbase/blob/51032ad14025d39405739345becbca37fab3cdb5/src/logservice/logminer/ob_log_miner_logger.cpp#L87
But I find we always print to `stdout`.

Reply:
> What happens if the user redirects the standard output to a file?

`ioctl` will return -1

Reply:
In that case, I think it's better not to output a warning log for scenarios where the output is redirected to a file. use`isatty(STDOUT_FILENO) `

Reply:
> https://github.com/oceanbase/oceanbase/blob/51032ad14025d39405739345becbca37fab3cdb5/src/logservice/logminer/ob_log_miner_logger.cpp#L87
> 
> 
> But I find we always print to `stdout`.

stdout will be redirect

Reply:
> In that case, I think it's better not to output a warning log for scenarios where the output is redirected to a file. use`isatty(STDOUT_FILENO) `

I get it. A little question: `LOG_WARN` or `LOG_WARN_RET` will output where and their difference. :)

Reply:
for oblogminer, the running log is: oblogminer.log.

Reply:
> for oblogminer, the running log is: oblogminer.log. The difference is that `LOG_WARN_RET` prints the first argument as an error code in the log and does not change the `ret`

Thank you for your answer. :)